receiver/scraperhelper/option.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package scraperhelper // import "go.opentelemetry.io/collector/receiver/scraperhelper"
+
+import "time"
+
+// Option allows for overriding the defaults in
+// a developer friendly way
+type Option interface {
```

Comment:
We can already set the settings manually on the struct since they are public fields, I don't think we should add a second way of doing the same thing; does this approach allow you to do anything you couldn't do before?

Reply:
This is more of a devx improvement to avoid having simplify setting default configuration.

An example of this can be seen here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/zookeeperreceiver/factory.go#L33-L34

Not many services set the value, but enough do that it would be nice to have this improvement.

Reply:
+1 to @mx-psi. I had the same comment. Though I think it is fine to remove the old way with type since is not needed, and we can add Options without having any option defined yet (since adding varargs later is still considered a breaking change).

Reply:
Just to be sure, @bogdandrutu are you saying that we should remove the `component.Type` parameter and leave the method as `NewDefaultScraperControllerSettings` ?